components/server/src/server.rs

```diff
@@ -1531,7 +1531,14 @@ impl ConfiguredRaftEngine for RocksEngine {
     fn register_config(&self, cfg_controller: &mut ConfigController, share_cache: bool) {
         cfg_controller.register(
             tikv::config::Module::Raftdb,
-            Box::new(DBConfigManger::new(self.clone(), DBType::Raft, share_cache)),
+            Box::new(DBConfigManger::new(
+                Arc::new(DummyRocksEngineFactory {
```

Comment:
Why not use v1 factory?

Reply:
v1 factory is simply cannot be built in this context. It requires KvEngineFactoryBuilder which will need other structs that are only available during initialization. 
On the other hand, v1 factory is designed to really create a db. But here the db already exists, and we just need a wrapper around it to unify the code.

Reply:
Factory also provides a way to query actual tablet, so accessing engine through factory is OK. I think it can be passed from initialization to config.

Reply:
Discussed offline. The best way is to extract a dedicate trait TabletAccessor for accessing tablets, either from single engine or multi-engines.